src/lib/server/response_builder.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <memory>
+#include "operators/abstract_operator.hpp"
+#include "postgres_protocol_handler.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+// The ResponseBuilder serializes the result data returned by Hyrise.
+class ResponseBuilder {
```

Comment:
Naming: It looks like it also writes data. A builder should not modify external state.